src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -776,12 +790,43 @@ export class SearchResult extends Disposable {
 		this._register(this.modelService.onModelAdded(model => this.onModelAdded(model)));
 
 		this._register(this.onChange(e => {
+			this._onRefreshSearchResult.fire(e);
 			if (e.removed) {
 				this._isDirty = !this.isEmpty();
 			}
 		}));
 	}
 
+	async performMassReplace(elementsToReplace: RenderableMatch[]) {
+		this._onRefreshSearchResult.pause();
+		await Promise.all(elementsToReplace.map(async (elem) => {
+			const parent = elem.parent();
+
+			if ((parent instanceof FolderMatch || parent instanceof FileMatch) && arrayContainsElementOrParent(parent, elementsToReplace)) {
+				// skip any children who have parents in the array
+				return;
+			}
+
+			if (elem instanceof FileMatch) {
+				elem.parent().replace(elem);
+			} else if (elem instanceof Match) {
+				elem.parent().replace(elem);
+			} else if (elem instanceof FolderMatch) {
+				await elem.replaceAll();
+			}
+		}));
+		this._onRefreshSearchResult.resume();
```

Comment:
Might be good to have a try/finally here to make sure the emitter gets unpaused if something goes wrong